pkg/server/node.go

```diff
@@ -1054,6 +1055,76 @@ func (n *Node) RangeFeed(
 	return nil
 }
 
+func (n *Node) UnsafeHealRange(
+	ctx context.Context, req *roachpb.UnsafeHealRangeRequest,
+) (_ *roachpb.UnsafeHealRangeResponse, rErr error) {
+	defer func() {
+		if r := recover(); r != nil {
+			rErr = errors.Errorf("%v", r)
+		}
+	}()
+
+	var desc roachpb.RangeDescriptor
+	var expValue *roachpb.Value
+
+	if err := n.storeCfg.DB.Txn(ctx, func(ctx context.Context, txn *kv.Txn) error {
+		kvs, err := sql.ScanMetaKVs(ctx, txn, roachpb.Span{
+			// TODO(tbg): paginate.
+			Key:    roachpb.KeyMin,
+			EndKey: roachpb.KeyMax,
+		})
+		if err != nil {
+			return err
+		}
+
+		for i := range kvs {
+			if err := kvs[i].Value.GetProto(&desc); err != nil {
+				return err
+			}
+			if desc.RangeID == req.Desc.RangeID {
+				expValue = kvs[i].Value
+				return nil
+			}
+		}
```

Comment:
@tbg We discussed how we wanted the higher level abstraction to include the RangeDescriptor in the RPC call, rather than just the RangeID. However, I think we still have to scan through all ranges to find the `expValue` as we need it for `CPut`. Do you think it makes sense to have the higher level abstraction also pass in `expValue`?

Reply:
Discussed in pod meeting today - a higher level operation will do the CPut and range desc retrieval, so this can all go for now.